array_api_strict/tests/test_array_object.py

```diff
@@ -112,12 +160,12 @@ def test_operators():
         "__mod__": "real numeric",
         "__mul__": "numeric",
         "__ne__": "all",
-        "__or__": "integer_or_boolean",
+        "__or__": "integer or boolean",
         "__pow__": "numeric",
         "__rshift__": "integer",
         "__sub__": "numeric",
-        "__truediv__": "floating",
-        "__xor__": "integer_or_boolean",
+        "__truediv__": "floating-point",
+        "__xor__": "integer or boolean",
```

Comment:
For my education, is there a reason for renaming these other than aesthetics? In particular the `integer_or_boolean` makes me wonder if there was a reason why it was created without spaces in it

Reply:
Totally not aesthetics! This is to match the keys from https://github.com/data-apis/array-api-strict/blob/main/array_api_strict/_dtypes.py#L117 , which are `"integer or boolean"` without underscores. Now that `_check_op_array_scalar` from `test_array_object.py` is reused in `test_elementwise_functions.py` and there is looping over the keys of the `_dtype_categories` dict, it saves a bit of gymnastics to map different spellings onto each other.

Reply:
Aha! That sounds totally sensible, less 🤸